proposals/selfAssemblingModelHierarchy/README.md

```diff
@@ -0,0 +1,208 @@
+#### Recommendation
+This proposal advocates #2 or #3 as being the best choice as they minimize impact to current usage. The only thing that would have to change is `kind=""` would have to be explicitly authored to explicitly stop propagation. The biggest risk to either of these usages is that code sites may be doing explicit `GetMetadata("kind")` calls and comparisons instead of using the `UsdModelAPI`. For clarity, some of these variations may benefit from the concept to author a more explicit model hierarchy block-- say `kind="none"`, but no proposal strictly requires introducing this.
+
+### Instancing
+Propagation when `instanceable=True` is complicated, as a prim may have multiple parents with different model hierarchy validity. There are actually [bugs](https://github.com/PixarAnimationStudios/USD/issues/2406) with this today.
+
+The USD development team proposes that prim flags are considered in concert with composition arcs when constructing prototypes to deal with this ambiguity.
+
+Until that fix is available, the best that can be done in both the current state of the world and with self propagating hierarchies is to encourage users to be intentional about model hierarchy when authoring `instanceable` to avoid violating the hierarchy's continuity.
+
+### Additional API
+
+To further simplify usage of model hierarchy for developers, this proposal also advocates introducing `IsAssembly`, `IsComponent`, and `IsSubcomponent` methods on `UsdModelAPI`, along with associated cached prim flags and traversal predicates.
+
+This proposal also recommends `IsModel` should be deprecated in favor of `IsModelHierarchy` and should be equivalent to (`IsGroup() || IsComponent()`).
+
+The `UsdPrimIsModel` predicate should be deprecated in favor of `UsdPrimIsModelHierarchy` and should be equivalent to `UsdPrimIsGroup || UsdPrimIsComponent`.
+
+The USD core API could also provide ranges to iterate explicitly over `assembly`, `component` or `subcomponent` prims, skipping over the intermediate prims. This might be easier to provide in C++20 (with the forthcoming ranges specification) and could be deferred until that point.
+
+## Summary
+This proposal aims to fufill the promise of self assembling model hierarchy as indicated in the USD documentation. If rendering behavior is going to depend on model hierarchy being correctly specified through path expressions, it's important to simplify correct assembly and maintenance of model hierarchy for users and tools.
+
+As the "East River Conundrum" demonstrates, there isn't a trade-off free solution to this problem, but importantly, the trade-offs can be largely mitigated for currently correctly specified model hierarchies and API usage.
+
+## Questions
+* This proposal rules out multi-pass population of prim data. Was that a correct assumption?
```

Comment:
We did special work for payloads to avoid two-pass composition, and yes we’d like to avoid two-pass population for similar performance reasons.  Also, the two-pass approach, while avoiding the **traversal** pathology of the “East River Conundrum”, still incurs the “read kind on potentially all prims during population” cost.

Reply:
Good point on another drawback of multi-pass.